gradle-multi-module/jmp-cloud-bank-impl/src/main/java/com/sungjun/cloud.bank.impl/CloudBankServiceImpl.java

```diff
@@ -0,0 +1,67 @@
+package com.sungjun.cloud.bank.impl;
+
+import com.sungjun.bank.api.BankApi;
+import com.sungjun.dto.entity.BankCard;
+import com.sungjun.dto.entity.BankCardType;
+import com.sungjun.dto.entity.CreditBankCard;
+import com.sungjun.dto.entity.DebitBankCard;
+import com.sungjun.dto.entity.UserEntity;
+import com.sungjun.dto.repository.CreditBankCardRepository;
+import com.sungjun.dto.repository.DeditBankCardRepository;
+import java.util.Random;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@AllArgsConstructor
+public class CloudBankServiceImpl implements BankApi {
+
+  private final DeditBankCardRepository deditBankCardRepository;
+  private final CreditBankCardRepository creditBankCardRepository;
+
+  @Override
+  public BankCard createBankCard(UserEntity user, BankCardType bankCardType) {
+    if (BankCardType.DEBIT.equals(bankCardType)) {
+      BankCard creditBankCard = getBankCard(user, bankCardType);
+      DebitBankCard card = deditBankCardRepository.save((DebitBankCard) creditBankCard);
+      return card;
+    } else if (BankCardType.CREDIT.equals(bankCardType)) {
+      BankCard creditBankCard = getBankCard(user, bankCardType);
+      CreditBankCard card = creditBankCardRepository.save((CreditBankCard) creditBankCard);
+      return card;
+    }
+    return null;
+  }
+
+  private BankCard getBankCard(UserEntity user, BankCardType bankCardType) {
+    BankCard creditBankCard;
+    if (BankCardType.CREDIT.equals(bankCardType)) {
+      creditBankCard = new CreditBankCard();
```

Comment:
We could have used method reference here

---

23. Reimplement createBankCard with method reference (ex: CreditBankCard::new).

Reply:
I didn't catch your idea. I don't see a way to use method reference here due to syntax error;